lib/ui/home.dart

```diff
@@ -22,7 +26,28 @@ class Home extends StatefulWidget {
   }
 }
 
-class _HomeState extends State<Home> {
+class _HomeState extends State<Home> with WidgetsBindingObserver {
+  HomeViewmodel homeViewModel;
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addObserver(this);
+  }
+
+  @override
+  void dispose() {
+    WidgetsBinding.instance.removeObserver(this);
+    super.dispose();
+  }
+
+  @override
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    if (state == AppLifecycleState.resumed) {
+      _retrieveDynamicLink();
```

Comment:
After opening a deeplink on the app, when I put the app in the backround then open again in the foreground, a new screen is opened. I can do the background to foreground switching several times and I'll get several screens. 

The app needs to differentiate between actual opening of a deeplink from a normal user switch to backround and not reprocess links.

Reply:
👍

Reply:
after investigation , the firebase lib call should be returning null on the second call, this seems to be a bug that's fixed in the latest version. @isaacmartinezimv